lib/engine/step/g_18_zoo/choose_power.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Engine
+  module Step
+    module ChoosePower
+      def choice_name
+        'Use powers'
+      end
+
+      def choices
+        entity = current_entity
+        @choices = Hash.new { |h, k| h[k] = [] }
+        @choices['midas'] = 'Midas' if company_midas?(entity)
+        @choices['holiday'] = 'Holiday' if company_holiday?(entity)
+        @choices['greek_to_me'] = 'It’s all greek to me' if company_greek_to_me?(entity)
+        @choices['whatsup'] = 'Whatsup' if company_whatsup?(entity)
+        @choices
+      end
+
+      def choice_available?(entity)
+        company_midas?(entity) || company_holiday?(entity) || company_whatsup?(entity) ||
+          company_greek_to_me?(entity)
+      end
+
+      def company_midas?(entity)
+        entity&.player? && entity.companies.any? { |c| c.sym == :MIDAS }
```

Comment:
do yo uhav ea refernece to midas? better to do entity.companiens.include?(midas)